flink-connector-jdbc-postgres/src/main/java/org/apache/flink/connector/jdbc/postgres/database/dialect/PostgresDialect.java

```diff
@@ -73,8 +74,10 @@ public Optional<String> getUpsertStatement(
                 Arrays.stream(uniqueKeyFields)
                         .map(this::quoteIdentifier)
                         .collect(Collectors.joining(", "));
+        final Set<String> uniqueKeyFieldsSet = new HashSet<>(Arrays.asList(uniqueKeyFields));
         String updateClause =
                 Arrays.stream(fieldNames)
+                        .filter(f -> !uniqueKeyFieldsSet.contains(f))
```

Comment:
This change makes sense to me after read Postgresql `ON CONFLICT ... DO UPDATE SET..` SQL grammar, could you add a unit test for this change ?

Reply:
Thanks for review. I have added a unit test.